extensions/pkg/webhook/cloudprovider/cloudprovider.go

```diff
@@ -61,6 +59,11 @@ func New(mgr manager.Manager, args Args) (*extensionswebhook.Webhook, error) {
 		Webhook:  &admission.Webhook{Handler: handler},
 		Path:     WebhookName,
 		Selector: namespaceSelector,
+		ObjectSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				v1beta1constants.GardenerPurpose: v1beta1constants.SecretNameCloudProvider,
+			},
+		},
```

Comment:
This is not backwards-compatible :( You probably need a field in `Args` to control whether the object selector should be enabled or not. Then extensions can set this field based on the Gardener version (only for shoots reconciled with Gardener > 1.41 the `cloudprovider` secret will get the label).